sign.go

```diff
@@ -0,0 +1,101 @@
+package floodsub
+
+import (
+	"fmt"
+
+	pb "github.com/libp2p/go-floodsub/pb"
+
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+const SignPrefix = "libp2p-pubsub:"
+
+func verifyMessageSignature(m *pb.Message) error {
+	pubk, err := messagePubKey(m)
+	if err != nil {
+		return err
+	}
+
+	xm := pb.Message{
```

Comment:
This is fine if we're never planning on adding additional fields. Otherwise, we should probably copy the *entire* message, remove just the signature, and remarshal that. That should keep the unknown fields.

Reply:
It's probably more futureproof to do it that way, although ... message copying again.

Reply:
It should be as efficient or more efficient than the current code. I'm suggesting:

```
xm := *m
xm.Signature = nil
xm.Key = nil
```

Reply:
Sure, we can do that.

Reply:
done.